pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -380,6 +383,38 @@ func NewNodeLifecycleController(podInformer coreinformers.PodInformer,
 	return nc, nil
 }
 
+func (nc *Controller) insertNodeToProcessing(nodeName string) bool {
+	nc.lock.Lock()
+	defer nc.lock.Unlock()
+	if nc.nodesInProcessing.Has(nodeName) {
+		return false
+	}
+	nc.nodesInProcessing.Insert(nodeName)
+	return true
+}
+
+func (nc *Controller) removeNodeFromProcessing(nodeName string) {
+	nc.lock.Lock()
+	defer nc.lock.Unlock()
+	nc.nodesInProcessing.Delete(nodeName)
+}
+
+// WARNING: If you're adding any return calls or defer any more work from this
+// function you have to make sure to update nodesInProcessing properly with the
+// disposition of the node when the work is done.
+func (nc *Controller) putNodeToUpdateChannel(node *v1.Node) error {
+	if node == nil {
+		return nil
+	}
+	if !nc.insertNodeToProcessing(node.Name) {
```

Comment:
If node is updating, we should also process the new one in next loop. Otherwise, the node condition may not be updated.

Reply:
I think that should be ok given that the node will enter processing later again on node heartbeat - so there should be eventual consistency.